easybuild/tools/options.py

```diff
@@ -1524,6 +1529,70 @@ def check_root_usage(allow_use_as_root=False):
                                  "so let's end this here.")
 
 
+def handle_include_easyblocks_from(options, log):
+    """
+    Handle --include-easyblocks-from-pr and --include-easyblocks-from-commit
+    """
+    def check_included_multiple(included_easyblocks_from, source):
+        """Check whether easyblock is being included multiple times"""
+        included_multiple = included_easyblocks_from & included_easyblocks
+        if included_multiple:
+            warning_msg = "One or more easyblocks included from multiple locations: %s " \
+                          % ', '.join(included_multiple)
+            warning_msg += "(the one(s) from %s will be used)" % source
+            print_warning(warning_msg)
+
+    if options.include_easyblocks_from_pr or options.include_easyblocks_from_commit:
+
+        if options.include_easyblocks:
+            # check if you are including the same easyblock twice
+            included_paths = expand_glob_paths(options.include_easyblocks)
+            included_easyblocks = set([os.path.basename(eb) for eb in included_paths])
```

Comment:
If `options.include_easyblocks` is not set (so no `--include-easyblock=/path/to/*.py`) then `included_easyblocks` is not set when we get to `included_easyblocks |= included_from_pr` (https://github.com/easybuilders/easybuild-framework/pull/4468/files#diff-6c50cde4279e48be792a6b7bc52a5cfce1c795608113dbc3b89a1ea528985d62R1565)

```log
$ eb --include-easyblocks-from-pr=3257 
== Temporary log file in case of crash /tmp/eb-bjr7462f/easybuild-fxppjtpl.log
Traceback (most recent call last):
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/simon/work/easybuild-framework/easybuild/main.py", line 768, in <module>
    init_session_state, eb_go, cfg_settings = prepare_main()
  File "/home/simon/work/easybuild-framework/easybuild/main.py", line 762, in prepare_main
    eb_go, cfg_settings = set_up_configuration(args=args, logfile=logfile, testing=testing)
  File "/home/simon/work/easybuild-framework/easybuild/tools/options.py", line 1704, in set_up_configuration
    handle_include_easyblocks_from(eb_go.options, log)
  File "/home/simon/work/easybuild-framework/easybuild/tools/options.py", line 1565, in handle_include_easyblocks_from
    included_easyblocks |= included_from_pr
UnboundLocalError: local variable 'included_easyblocks' referenced before assignment
```

```
eb --include-easyblocks-from-pr=3257 --include-easyblocks=/tmp/easyblocks/generic/*.py
== Temporary log file in case of crash /tmp/eb-sayqk6s7/easybuild-7xc3jvxm.log
== easyblock quantumespressocmake.py included from PR #3257
Usage: eb [options] easyconfig [...]
```

Reply:
fixed in [52648c0](https://github.com/easybuilders/easybuild-framework/pull/4468/commits/52648c09a33c17885ee0fb73847dbe9ef2aadb2a)